### ISSUE TITLE
[bug] Raw values in variants don't override presets

### ISSUE THREAD

**Issue by nandorojo** [tamagui/tamagui#293]
**Describe the bug**

```ts
const Frame = styled(ButtonFrame, {
  variants: {
    square: {
      ':number': (size, { tokens }) => {
        return {
          width: tokens.size[size] ?? size,
          p: '$0',
        }
      },
    },
  } as const,
})
```

This works as expected. However, if I change `p: '$0'` → `p: 0` (or any number), then padding no longer overrides. Instead, it falls back to the default values set by tamagui's button frame.

Here is the output CSS for `p: '$0'`:

<img width="406" alt="Screen Shot 2022-10-20 at 1 10 36 PM" src="https://user-images.githubusercontent.com/13172299/197014399-d9f3d9c8-7ae5-43d6-8ee7-8e851532ce6c.png">

And if I do `p: 0`, I see this instead. Notice that padding does not override.

<img width="384" alt="Screen Shot 2022-10-20 at 1 12 28 PM" src="https://user-images.githubusercontent.com/13172299/197014708-a460123f-6ced-4d2d-bf4b-863894b486a1.png">

If I do `pl: 0, pr: 0`, it also doesn't get fixed.

In order to fix it, I need to explicitly pass `paddingRight: 0, paddingLeft: 0`. It looks like the objects might be getting merged before the `pl` shorthand is turned into `paddingLeft`, such that `pl` doesn't override the default `paddingLeft`. 

Meanwhile, `p: '$0'` _does_ seem to override it. However, I think this behavior is actually _incorrect_, since `p` is less specific than `paddingLeft`, and React Native styling is based on styling specificity. My assumption is that the compiler is turning `p` into `paddingLeft`, `paddingRight`, ...etc, but I think this may be breaking the RN specificity order.

Happy to discuss this further as I feel I understand the issue.

**To Reproduce**

Lmk if I need a full repro. Here's the component code:

```tsx
import { forwardRef } from 'react'
import {
  ButtonFrame,
  ButtonText,
  ButtonProps,
  useButton,
  styled,
  TamaguiElement,
} from 'tamagui'

const Frame = styled(ButtonFrame, {
  variants: {
    square: {
      ':number': (size, { tokens }) => {
        return {
          width: tokens.size[size] ?? size,
          p: '$0',
        }
      },
    },
  } as const,
})

const Text = styled(ButtonText, {})

export const Button = forwardRef<
  TamaguiElement,
  ButtonProps & {
    square?: boolean
  }
>(function Btn(p, ref) {
  const { props } = useButton(p, {
    Text,
  })

  return <Frame {...(props as any)} ref={ref} square={p.square && p.size} />
})
```

And then just do:

```tsx
<Button square />
```

**System Info (please complete the following information):** 

Output of `npx envinfo --system --npmPackages --binaries --browsers`

**Tamagui info**

The output here was too big for the terminal to be able to handle. I don't think the `!debug` is working for me, does this require using the compiler?

- [ ] yarn.lock or similar
- [ ] If it's a specific file, add `//! debug` to the top (the `!` works in esbuild compiled packages but not necessary always) and include that (only for compiler, but helps to know if it's on or off).
- [ ] Run your server/bundler with `DEBUG=tamagui` and include that output at the bottom.

**Comment by natew** [comment]
I can see this - `defaultVariants` is adding after props of parent.

**Comment by natew** [comment]
Fixed in 280a49db94305efe0e231b3871c09b27d1935f6f
